scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -317,15 +356,46 @@ static void dom_load_xfer_task(struct task_struct *p, struct task_ctx *taskc,
 	bpf_spin_unlock(&to_lockw->lock);
 
 	if (debug >= 2)
-		bpf_printk("XFER dom%u->%u task=%lu from=%lu->%lu to=%lu->%lu",
-			   from_dom_id, to_dom_id,
+		bpf_printk("XFER DCYCLE dom%u->%u task=%lu from=%lu->%lu to=%lu->%lu",
+			   from_domc->id, to_domc->id,
 			   task_dcycle >> RAVG_FRAC_BITS,
 			   from_dcycle[0] >> RAVG_FRAC_BITS,
 			   from_dcycle[1] >> RAVG_FRAC_BITS,
 			   to_dcycle[0] >> RAVG_FRAC_BITS,
 			   to_dcycle[1] >> RAVG_FRAC_BITS);
 }
 
+static u64 dom_min_vruntime(struct dom_ctx *domc)
+{
+	struct lock_wrapper* lockw;
+	u64 min_vruntime;
+
+	lockw = lookup_dom_vtime_lock(domc->id);
+	if (!lockw) {
+		return 0;
+	}
+
+	bpf_spin_lock(&lockw->lock);
+	min_vruntime = domc->min_vruntime;
```

Comment:
The locking around is unlikely to make any difference when reading a single value.

Reply:
So, the reason I did this is that _technically_ what we were doing before is undefined behavior according to the C standard:

>The execution of a program contains a data race if it contains two conflicting actions in
>different threads, at least one of which is not atomic, and neither happens before the
>other. Any such data race results in undefined behavior.

In practical terms this is of course unlikely to be a problem as long as the compiler doesn't do anything annoying like issue torn writes or reads. Given that we've gone this far without synchronizing, I can just simplify and get rid of the extra lock. Maybe in a follow-on change I can just add `SCX_READ_ONCE()` and `SCX_WRITE_ONCE()` macros to match the LKMM?